src/adapt_tests.cpp

```diff
@@ -49,11 +49,11 @@ void test_adapt(parser const &problem, std::filesystem::path gold_base)
   auto const table_coarse_path = gold_base / (prefix + "coarse_table.dat");
 
   auto const fval_orig =
-      fk::vector<P>(read_vector_from_txt_file(fval_orig_path));
+      fk::vector<P>(read_vector_from_txt_file<P>(fval_orig_path));
   auto const gold_coarse =
-      fk::vector<P>(read_vector_from_txt_file(fval_coarse_path));
+      fk::vector<P>(read_vector_from_txt_file<P>(fval_coarse_path));
```

Comment:
```suggestion
      read_vector_from_txt_file<P>(fval_coarse_path);
```
`read_vector_from_txt_file` also has an extra unnecessary move constructor.